### PR TITLE
autonomous docker Compose file

### PR DESCRIPTION
**Pull request by michalmikolajczyk**
This PR provides the minimum code required, to start a dockerized, basic instance of Sharetribe, including MySQL database, a `Web` and a `Worker` process of the app, and memcached. It allows to start new instances, as well as work with existing marketplaces, by importing the database.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 services:
+  mysql:
+    image: mysql:5.7
+    ports:
+      - "3306:3306"
+    environment:
+      MYSQL_USER: sharetribe
+      MYSQL_PASSWORD: secret
+      MYSQL_DATABASE: sharetribe_production
+      MYSQL_ROOT_PASSWORD: secret
```
Comment (MikeLuDev) [inline docker-compose.yml:11]: Should these be stored in `.env`?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+
   web:
     build: .
     ports:
       - "3000:3000"
     env_file: .env
+    links:
+      - mysql:mysql
+    depends_on:
+      - mysql
```
Comment (MikeLuDev) [inline docker-compose.yml:19]: Pretty sure you can use either `links` or `depends_on`, but both are not needed.

https://stackoverflow.com/questions/35832095/difference-between-links-and-depends-on-in-docker-compose-yml

**Review by MikeLuDev — CHANGES_REQUESTED** [review]
Tested locally, seems to work. Few small changes.

File: docker-compose.yml (hunk 1 above)
Comment (michalmikolajczyk) [reply docker-compose.yml:11]: not necessarily. I was thinking about that, seems like a zero-sum game.

File: docker-compose.yml (hunk 2 above)
Comment (michalmikolajczyk) [reply docker-compose.yml:19]: that's done, I removed `links`, across the Compose file

**Comment by michalmikolajczyk** [comment]
@tksventures/engineers this one is ready for review

File: kubernetes-compose.yml [hunk 3]
```diff
@@ -0,0 +1,234 @@
+apiVersion: v1
+kind: List
```
Comment (bravetarget) [inline kubernetes-compose.yml:2]: Is using a `List` of multiple resources result in the same behavior as separating the items by `---`?

File: kubernetes-compose.yml [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    SECRET_KEY_BASE: IjFlZmQ3N2FhMmY2ZGVhZDg0MTJlNjg2Zjk4OWIwZjI3YTJhNDI3YjNiODY2NjkzYjUzMWQzNGZjNTlhMzMxMzVlODU0YmQ2NjBhZDQ5M2I0OTU3ZGJlYmQ0MTA3ZTNhZTgxODg3ZTI4YjE3MThlMDcwZWExMzU4M2IxZWUxOTMwIg==
+  kind: Secret
+  metadata:
+    name: sharetribe-secret
+  type: Opaque
+- apiVersion: v1
+  data:
+    MYSQL_DATABASE: c2hhcmV0cmliZV9wcm9kdWN0aW9u
+    MYSQL_PASSWORD: c2VjcmV0
+    MYSQL_ROOT_PASSWORD: c2VjcmV0
```
Comment (bravetarget) [inline kubernetes-compose.yml:79]: assuming these are placeholder values but just curious, is the encryption used for secrets easily decryptable?  if someone was to import into this into their own cluster, would they see the original values?

File: kubernetes-compose.yml [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            privileged: false
+            readOnlyRootFilesystem: false
+            runAsNonRoot: false
+          stdin: true
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+          tty: true
+
+        # mysql
+        - envFrom:
```
Comment (bravetarget) [inline kubernetes-compose.yml:189]: have been formatting containers to start with their `name` and `image` definitions first, has helped for readability; comment works too!

**Review by bravetarget — APPROVED** [review]
Configuration looks great! 👍

**Review by MikeLuDev — APPROVED** [review]

File: kubernetes-compose.yml (hunk 4 above)
Comment (MikeLuDev) [reply kubernetes-compose.yml:79]: Yes, I believe what you described would be the case. They could see/modify the original values.

**Pull request merged**